module/VuFind/src/VuFind/Config/ConfigManager.php

```diff
@@ -51,25 +53,33 @@
  */
 class ConfigManager
 {
-    use MergeRecursiveTrait;
+    use KeyGeneratorTrait;
 
     /**
-     * Configuration cache
+     * Default cache (Required to avoid warnings using the KeyGeneratorTrait).
+     */
+    protected StorageInterface $cache;
+
+    /**
+     * Simple cache to avoid accessing the advanced cache multiple times per request.
      *
-     * @var array[]
+     * @var array
      */
-    protected array $cache = [];
+    protected array $loadedConfig = [];
```

Comment:
It seems potentially redundant to have an in-memory cache at both the ConfigManager and ConfigLoader layers; doesn't that potentially mean we'll be double-storing some configs? Might it make sense to add a setter for the ConfigLoader's in-memory cache, and use that from here, so there's only one in-memory cache, or do the different cache key structures make this impractical/impossible?

Reply:
I agree, I think this is okay, since the loader already has force-reload capabilities built in. And if there's any worry about getting cached cache configs out of the cache (there's a tongue-twister!), that is of course impossible because we have to load the config before the cache can exist. :-)